### PR TITLE
[misc] migrate to the native mongo driver

### PR DESCRIPTION
**Pull request by das7pad**
### Description

For https://github.com/overleaf/issues/issues/2907
This PR is migrating the mongo driver from mongojs to mongodb -- the native driver.

#### Related Issues / PRs

https://github.com/overleaf/issues/issues/2907
Previous migration: https://github.com/overleaf/chat/pull/53 https://github.com/overleaf/chat/pull/54 https://github.com/overleaf/docstore/pull/66 (https://github.com/overleaf/contacts/pull/49 https://github.com/overleaf/document-updater/pull/144 https://github.com/overleaf/github-sync/pull/57 https://github.com/overleaf/notifications/pull/27 https://github.com/overleaf/project-history/pull/358 pending in review)

### Review

<details>
<summary> changed startup </summary>

The startup behavior has been changed in the chat service already https://github.com/overleaf/chat/pull/54 and is running with success in prod by now.

> This PR changes the startup of the service to not accept any http requests until after the mongo connection has been established. It also adds the correct way to handle connection setup errors, to catch and log them -- vs just letting them bubble up as unhandled PromiseRejections.
> 
> The web service can start without an established/working mongo connection, which is bad. Eventually we can port this snippet over.

</details>

#### Potential Impact

High. All interactions which persist changes to mongo.

#### Manual Testing Performed

- create a new project
- enable spell check
- write two typoed words
- see the two words with red underline
- add both words to the dics
- reload the page
- no red underline for the words
- in the admin panel -> select user -> dict -> delete one word
- reload the page and see one word only
- open project and see the deleted words with red underline
- add the word back in
- in the admin panel -> user -> dict -> delete the full dict
- restart spelling (caching issue)
- reload the admin page
- see an empty dict
- open the project
- see both words with red underline
- testing complete :)

### CODE REVIEW

**Review by gh2k — APPROVED** [review]
approval

**Comment by das7pad** [comment]
I've added a similar commit as overleaf/chat#60 to this PR.

File: app.js [hunk 1]
```diff
@@ -44,12 +45,20 @@ const port = settings && settings.port ? settings.port : 3005
...
-    return logger.info(`spelling starting up, listening on ${host}:${port}`)
-  })
+  mongodb
+    .waitForDb()
+    .then(() => {
+      app.listen(port, host, function (error) {
+        if (error != null) {
+          throw error
+        }
+        return logger.info(`spelling starting up, listening on ${host}:${port}`)
```
Comment (gh2k) [inline app.js:55]: Unnecessary return?

**Review by gh2k — APPROVED** [review]
Looks good. I noticed one unnecessary return that won't need re-review if you remove it.

File: app.js (hunk 1 above)
Comment (das7pad) [reply app.js:55]: Yes. There are quite a lot of others too. I just changed the indent only here -- or the collection method in other places. The spelling service is due for a general decaff-cleanup -- or removal after all, hint front end spell-check hint, hint :)

**Pull request merged**